path.go

```diff
@@ -55,6 +76,133 @@ func NewPathFromSVGString(svg string) (*Path, error) {
 	return p, nil
 }
 
+func newPathFromSvgPath(svgPath svg.SvgPath) (*Path, error) {
+	p := &Path{}
+
+	p.createPath(svgPath)
+	if err := p.createFillPaint(svgPath); err != nil {
+		return nil, err
+	}
+	if err := p.createStrokePaint(svgPath); err != nil {
+		return nil, err
+	}
+
+	runtime.SetFinalizer(p, func(obj *Path) {
+		ReleaseOnUIThread(func() {
+			skia.PathDelete(obj.path)
+		})
+	})
+	return p, nil
+}
+
+// createPath creates a skia path from an svg.Path.
+//
+// The coordinates used in svg.Operation are of type fixed.Int26_6, which has a fractional
+// part of 6 bits. When converting to a float the values are divided by 64.
+func (p *Path) createPath(svgPath svg.SvgPath) {
+	p.path = skia.PathNew()
+
+	for _, op := range svgPath.Path {
+		switch op := op.(type) {
+		case svg.OpMoveTo:
+			p.MoveTo(float32(op.X)/64.0, float32(op.Y)/64.0)
+		case svg.OpLineTo:
+			p.LineTo(float32(op.X)/64.0, float32(op.Y)/64.0)
+		case svg.OpQuadTo:
+			p.QuadTo(
+				float32(op[0].X)/64.0, float32(op[0].Y)/64.0,
+				float32(op[1].X)/64.0, float32(op[1].Y)/64.0,
+			)
+		case svg.OpCubicTo:
+			p.CubicTo(
+				float32(op[0].X)/64.0, float32(op[0].Y)/64.0,
+				float32(op[1].X)/64.0, float32(op[1].Y)/64.0,
+				float32(op[2].X)/64.0, float32(op[2].Y)/64.0,
+			)
+		case svg.OpClose:
+			p.Close()
+		}
+	}
+
+	if svgPath.Style.Transform != svg.Identity {
+		p.Transform(
+			geom.Matrix[float32]{
+				ScaleX: float32(svgPath.Style.Transform.A),
+				SkewX:  float32(svgPath.Style.Transform.C),
+				TransX: float32(svgPath.Style.Transform.E),
+				SkewY:  float32(svgPath.Style.Transform.B),
+				ScaleY: float32(svgPath.Style.Transform.D),
+				TransY: float32(svgPath.Style.Transform.F),
+			})
+	}
+}
+
+func (p *Path) createFillPaint(svgPath svg.SvgPath) error {
+	if svgPath.Style.FillerColor == nil ||
+		svgPath.Style.FillOpacity == 0 {
+		return nil
+	}
+
+	p.fillPaint = NewPaint()
+	p.fillPaint.SetStyle(paintstyle.Fill)
+
+	if color, ok := svgPath.Style.FillerColor.(svg.PlainColor); ok {
+		alpha := float32(color.A) / 0xFF
+		alpha *= float32(svgPath.Style.FillOpacity)
+		p.fillPaint.SetColor(ARGB(alpha, int(color.R), int(color.G), int(color.B)))
+	} else {
+		return fmt.Errorf("unsupported path fill style %T", svgPath.Style.FillerColor)
+	}
+
+	return nil
+}
+
+func (p *Path) createStrokePaint(svgPath svg.SvgPath) error {
+	if svgPath.Style.LinerColor == nil ||
+		svgPath.Style.LineOpacity == 0 ||
+		svgPath.Style.LineWidth == 0 {
+		return nil
+	}
+
+	strokeCap, ok := strokeCaps[svgPath.Style.Join.TrailLineCap]
+	if !ok {
+		return fmt.Errorf("unsupported path stroke cap %s", svgPath.Style.Join.TrailLineCap)
```

Comment:
In general, all errors in unison should be created with one of the functions in my `errs` package so that stack traces are attached. i.e. here, you could use `errs.Newf()`. I won't bother commenting on other places I find this, but would like to see them all replaced. Thanks.

Reply:
I have now used your `errs` package to create all errors.